server/src/main/java/org/apache/kafka/network/SocketServerConfigs.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.network;
+
+import org.apache.kafka.common.network.ListenerName;
+import org.apache.kafka.common.security.auth.SecurityProtocol;
+import org.apache.kafka.server.config.ReplicationConfigs;
+import java.util.Arrays;
+import java.util.stream.Collectors;
+
+public class SocketServerConfigs {
```

Comment:
Should we call it `ServerSocketConfigs` so as to align it with `ServerLogConfigs`?

Reply:
I name it `SocketServerConfigs `  instead of  `ServerSocketConfigs` for the following reasons
1.  this is part of `server.network`
2.  we have `SocketServe` which will move at some point to same package. 
3. java has something `java.net.ServerSocket` so we may need to differentiate between classes related `SocketServe` away from build in class. 

Am not very attached to the name but it made more sense for me during the refactor. WDYT?

Reply:
> Am not very attached to the name but it made more sense for me during the refactor. WDYT?

that makes sense to me. +1 :)